### DISCUSSION TITLE
cx_Freeze licensing  / frozen applications

### DISCUSSION THREAD

**Discussion by cainesi** [marcelotduarte/cx_Freeze#1613] [General]
@marcelotduarte, what is the expectation regarding how the cx-freeze license applies to frozen applications?

A frozen application contains a small amount of the cx-freeze code (in particular, the "base" that is used to load and start python, and some of the code from cx_Freeze/initscripts). Technically, I think that means the cx_Freeze license could apply to that frozen application.  The only consequences is that the frozen application should include a copy of the cx_Freeze license---but I doubt most users realize that!

I can see 3 possible solutions:
1) Include a note about this in the cx-freeze documentation / FAQ.
2) Have cx-freeze automatically include a copy of the cx-freeze license in the frozen application (maybe with some initial words to make it clear that the license only covers the particular bootstrap  code included from cx_freeze).  In that case, we could include an explanation in the documentation that this is being done.
3) We conclude that the license does not apply to frozen applications, and forget about this.

If you want to do either option 1 or 2, I can prepare a pull request for review.

**Comment by marcelotduarte** [comment]
Hi! 
As cx_Freeze uses the same license as python, I hope it applies in the same way, including the parts that are included, as you mentioned. I include myself in that majority who don't realize what part of cx_Freeze is included, despite having already modified that part a lot.
For the solutions presented, I prefer option 2, and I think some use the .dist-info/LICENSE of each package to include licenses, so we could force copy cx_Freeze-*.dist-info (they are in library.zip ). But in that case, it wouldn't have the initial words, it would be a rough copy. And I think mentioning that making it clear that the license only covers the specific bootstrap included from cx_freeze is also interesting.
@anthony-tuininga What do you think?

**Comment by cainesi** [comment]
PR #1672 has a potential implementation of option 2.  Happy to change it, if you think there is a better way.
